src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,802 @@
+                onlyLogCriticalEvents: !Debug,
+                loadProjectsReadOnly: true);
+
+            var failedBuildSubmissions = new ConcurrentBag<BuildSubmission>();
+
+            try
+            {
+                var sw = Stopwatch.StartNew();
+
+                var evaluationContext = EvaluationContext.Create(EvaluationContext.SharingPolicy.Shared);
+
+                ProjectGraph projectGraph;
+
+                int buildCount = 0;
+
+                try
+                {
+                    var buildParameters = new BuildParameters(projectCollection)
+                    {
+                        Loggers = new List<ILogger>
+                        {
+                            loggingQueue
+                        },
+                        OnlyLogCriticalEvents = false
+                    };
+
+                    BuildManager.DefaultBuildManager.BeginBuild(buildParameters);
+
+                    projectGraph = new ProjectGraph(entryProjects, projectCollection, (path, properties, collection) =>
+                    {
+                        var projectOptions = new ProjectOptions
+                        {
+                            EvaluationContext = evaluationContext,
+                            GlobalProperties = properties,
+                            LoadSettings = ProjectLoadSettings.IgnoreEmptyImports | ProjectLoadSettings.IgnoreInvalidImports | ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition | ProjectLoadSettings.IgnoreMissingImports,
+                            ProjectCollection = collection,
+                        };
+
+                        var project = Project.FromFile(path, projectOptions);
+
+                        var projectInstance = project.CreateProjectInstance(ProjectInstanceSettings.None, evaluationContext);
+
+                        if (properties.ContainsKey("TargetFramework") || string.IsNullOrWhiteSpace(projectInstance.GetPropertyValue("TargetFrameworks")))
+                        {
+                            BuildManager.DefaultBuildManager
+                                .PendBuildRequest(
+                                    new BuildRequestData(
+                                        projectInstance,
+                                        TargetsToBuild,
+                                        hostServices: null,
+                                        BuildRequestDataFlags.SkipNonexistentTargets))
+                                .ExecuteAsync(
+                                    callback: buildSubmission =>
+                                    {
+                                        if (buildSubmission.BuildResult.OverallResult == BuildResultCode.Failure)
+                                        {
+                                            failedBuildSubmissions.Add(buildSubmission);
+                                        }
+                                    },
+                                    context: null);
```

Comment:
Will these calls be awaited? Is the line BuildManager.DefaultBuildManager.EndBuild(); waiting for them?